### ISSUE TITLE
Produce post-mortem for HE.net outage in Amsterdam - 15 December 2024

### ISSUE THREAD

**Issue by Firefishy** [openstreetmap/operations#1196]
We has a full service outage from HE.net from approximately 15 December 2024 3:51am until 18 December 2024 00:29am in Amsterdam.

Our equipment continued running. Fibre links remained up, but upstream HE.net equipment failed. We lost all internet connectivity.

* 15 December 2024 03:53 - HE.net Outage starts.
* 15 December 2024 04:18 - OSM Ops Initial Email to HE.net NOC, asking if unplanned maintenance.
* 15 December 2024 04:24 - HE.net email response from HE.net NOC, not maintenance. Investigating issue. (Investigate why email receipt delayed to osmfoundation.org email)
* 15 December 2024 04:43 - OSM Ops phone HE.net for status update. Response paraphrase: "Outage in Amsterdam, we investigating. No Estimated time to recover. Equinix remote hands on-route" 
* 15 December 2024 07:24 - OSM Ops ask for Estimated time to recover.
* 15 December 2024 11:12 -  OSM Ops phone HE.net for status update. Response paraphrase: "No Estimated time to recover"
* ...
* 17 December 2024 12:21 - New Equinix Internet up and running. OpenStreetMap.org services restored. (56 hours)
* 18 December 2024 00:29 - HE.net confirm restoration of service. (68 hours outage)
TBC.

**Comment by drolbr** [comment]
FYI: Last known good connection at 03:53:15, first failed connection at 03:53:24:
````
2024-12-15 03:53:15 URL:https://osm-planet-eu-central-1.s3.dualstack.eu-central-1.amazonaws.com/planet/replication/minute/state.txt [86/86] -> "/srv/overpass/diffs/state.txt"
````
The asked-for server is `planet.openstreetmap.org`, then usually redirected.

**Comment by flohoff** [comment]
Just as a Datapoint - no clue what HE said the cause was or what you gathered yourself. 2024-12-16 08:01 (CET) i had a look at a HE BGP Looking Glass for the route to planet.openstreetmap.org - at that time - 184.104.179.145.  I have a screenshot of that output from core2.ams1.he.net showing 6 paths with an age of11 Days.
At the same time i did an "mtr" to the same address which was dropped on the first HE hop in my path in Düsseldorf, which i found astonishing. It may be that HE uses MPLS for forwarding and has ttl propagation off. Otherwise my guess would have been that some filtering on HEs routers went bad for that  184.104.176.0/21 on their border routers.

**Comment by pnorman** [comment]
We know it was upstream HE equipment that required a hardware replacement - the exact reasons of the HE failure shouldn't be the focus of the post-mortem, it needs to look at stuff within our control.

**Comment by Firefishy** [comment]
FYI we've received 3 day worth of credit from HE.net for the SLA breach.

**Comment by mmd-osm** [comment]
While read only access to the API was restored rather quickly (I believe around 8 hours or so), osm.org as an Identity provider was unavailable for the full 56 hour period. This had some impact other sites, like the community forum, umap editing and probably quite a few others that are relying on osm.org to be around.

Perhaps this topic deserves some further thought re. follow up activities.

**Comment by tomhughes** [comment]
What sort of "thought" do you think would be useful? Do you have some magical idea of how to make OAuth not need to write to the database?

**Comment by mmd-osm** [comment]
I'm thinking more about the lines that the IdP part is (maybe too) tightly coupled to a 13TB database. We had map edits that weren't replicated yet, causing potential inconsistencies for osmdbt and logical replication. In short, the fail over isn't exactly trivial. I'm wondering if splitting out the IdP part could improve the situation.

**Comment by AntonKhorev** [comment]
Is it difficult to let OAuth write to whatever tables it needs to and after the incident is over discard these writes?

**Comment by tomhughes** [comment]
It's completely impossible - when using WAL replication the entire database is either a master or a slave it can't be part and part and if it's a slave you can't write to it.

**Comment by mmd-osm** [comment]
Maybe it's a bit early to start looking for solutions while the analysis is still underway. Anyway...

Doorkeeper lets you define your own _access token_, _access grant_ and _application_ classes. Furthermore, when using the Rails [multi db](https://guides.rubyonrails.org/active_record_multiple_databases.html) feature, you can direct  database requests to another dedicated db instance. ([example implementation](https://github.com/doorkeeper-gem/doorkeeper/issues/1464#issuecomment-774459802))

This might be easier to manage during a fail over, since it is a much smaller db, and the impact of losing some tokens due to incomplete replication is probably negligible, compared to minutely diffs.

**Comment by tomhughes** [comment]
Yes and as soon as you do that you lose the ability to police foreign keys between those records and user records in the main database.

**Comment by Firefishy** [comment]
The post-mortem finally: https://github.com/openstreetmap/owg-website/pull/138

**Comment by Firefishy** [comment]
https://operations.osmfoundation.org/2025/02/15/post-mortem.html
